src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -466,19 +484,19 @@ class NBDWatchCtx : public librbd::UpdateWatchCtx
   }
 };
 
-static int open_device(const char* path, bool try_load_module = false)
+static int open_device(const char* path, Config *cfg = NULL, bool try_load_module = false)
```

Comment:
I add this default value for do_unmap, which will call open_device. I tested and not crash